File: core/src/androidTest/java/com/emarsys/core/util/HeaderUtilsInstrumentationTest.java
package com.emarsys.core.util;

import com.emarsys.testUtil.TimeoutUtils;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TestRule;

import static org.junit.Assert.assertEquals;

public class HeaderUtilsInstrumentationTest {

    private final String username = "user";

    @Rule
    public TestRule timeout = TimeoutUtils.getTimeoutRule();

    @Test(expected = IllegalArgumentException.class)
    public void testCreateBasicAuth_usernameShouldNotBeNull(){
        HeaderUtils.createBasicAuth(null);
    }

    @Test
    public void testCreateBasicAuth_shouldCreateCorrectBasicAuthString() throws Exception {
        String expected = "Basic dXNlcjo=";
        String result = HeaderUtils.createBasicAuth(username);
        assertEquals(expected, result);
    }

}

File: core/src/androidTest/java/com/emarsys/core/util/HeaderUtilsInstrumentationTest.kt
package com.emarsys.core.util

import com.emarsys.testUtil.AnnotationSpec
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.matchers.shouldBe

class HeaderUtilsInstrumentationTest : AnnotationSpec() {
    private val username = "user"

    @Test
    fun testCreateBasicAuth_usernameShouldNotBeNull() {
        shouldThrow<IllegalArgumentException> {
            HeaderUtils.createBasicAuth(null)
        }
    }

    @Test
    fun testCreateBasicAuth_shouldCreateCorrectBasicAuthString() {
        val expected = "Basic dXNlcjo="
        val result = HeaderUtils.createBasicAuth(username)
        result shouldBe expected
    }
}